src/native/libs/System.Security.Cryptography.Native/pal_evp_pkey_rsa.c

```diff
@@ -132,17 +132,6 @@ int32_t CryptoNative_RsaDecrypt(EVP_PKEY* pkey,
         goto done;
     }
 
-    // This check may no longer be needed on OpenSSL 3.0
```

Comment:
I don't know that we're off of OpenSSL < 3.0 everywhere.  Are we really ready to remove this?

Reply:
I've [heard reports](https://github.com/dotnet/runtime/issues/92798) that some ENGINEs don't set external key flags - previously I rejected that to be ENGINE implementation bug but now I'm really further convinced this should be removed overall because providers (at least `tpm2`) also don't do that.

Reply:
We very specifically added the checking of the external flag at the suggestion of a customer in https://github.com/dotnet/runtime/issues/53345#issuecomment-850560118 to work around an issue they were having with a PKCS#11 ENGINE.

If we are going to regress that scenario, then I think we should keep the external flag checking.

Reply:
And we are far from being able to drop OpenSSL 1.1.1.

Reply:
If the plan is to just not check if there is no private key at all and just let OpenSSL do what it is going to do, I suppose. I don't see how it is relevant to provider support or why we needed to make this change now.

Reply:
So currently `HasNoPrivateKey` rejects all private keys coming from provider because that flag is not set and parameters are also not set - I can add it back for OpenSSL < 3.0 if you believe that is segfaulting there.

I'm not sure I understand 'If we are going to regress that scenario, then I think we should keep the external flag checking' - I'm confused if you suggest to keep the existing logic or change it

Reply:
I honestly don't have the full context on it, other than Jeremy's comment here:

https://github.com/dotnet/runtime/issues/53345#issuecomment-852264901

The whole reason the check is there, from my understanding, is "letting OpenSSL handle it" means "segfault" for some version range of OpenSSL.

I also do not know if we even have CI protection using OpenSSL 1.1. anymore, or what versions are being used.

Reply:
I added the change for RSA back. For ECDSA/ECDH I'm not sure how to write that (at least not without further reading) and don't have setup for OSSL 1.1 so if any of you know how to quickly write that I can add the check just in case. Otherwise I'll open issue to finish this investigation within 9.0 timeline

Reply:
I created https://github.com/dotnet/runtime/issues/105175 to investigate it - I will be offline for the next two weeks so if any of you is interested feel free to pick it up (but do respond on the issue).